src/persist/src/indexed/runtime.rs

```diff
@@ -62,23 +70,39 @@ where
     let indexed = Indexed::new(log, blob, metrics.clone())?;
     // TODO: Is an unbounded channel the right thing to do here?
     let (tx, rx) = crossbeam_channel::unbounded();
+    let runtime_impl_config = config.clone();
     let runtime_impl_metrics = metrics.clone();
     let runtime_f = move || {
         // TODO: Set up the tokio or other async runtime context here.
-        let mut l = RuntimeImpl {
-            indexed,
-            rx,
-            metrics: runtime_impl_metrics,
-        };
+        let mut l = RuntimeImpl::new(runtime_impl_config, indexed, rx, runtime_impl_metrics);
         while l.work() {}
     };
     let id = RuntimeId::new();
-    let handle = thread::Builder::new()
+    let impl_handle = thread::Builder::new()
         .name(format!("persist-runtime-{}", id.0))
         .spawn(runtime_f)?;
-    let handle = Mutex::new(Some(handle));
+    let ticker_tx = tx.clone();
+    let ticker_handle = thread::Builder::new()
+        .name(format!("persist-ticker"))
+        .spawn(move || {
+            let tick_interval = config.min_step_interval / 10;
```

Comment:
I'm not really sure about the / 10 here. It seems like it gets entirely defeated by the fact that we check whether the `min_step_interval` has elapsed since the last step and so this makes us send 9 additional `tick`s that are no-ops. I think the intent is to provide a stronger guarantee that `step` will definitely get called within the requested interval, but it's not clear to me that dividing by 2 is any worse than dividing by 10 (because of the check mentioned above). At the very least the rationale here could use a comment